contracts/CompoundAssetProxy.sol

```diff
@@ -0,0 +1,143 @@
+        );
+    }
+
+    /// @notice Makes the actual ctoken deposit
+    /// @return Tuple (the shares minted, amount underlying used)
+    function _deposit() internal override returns (uint256, uint256) {
+        // Load balance of contract
+        uint256 amount = token.balanceOf(address(this));
+
+        // Since ctoken's mint function returns sucess codes
+        // we get the balance before and after minting to calculate shares
+        uint256 beforeBalance = ctoken.balanceOfUnderlying(address(this));
+
+        // Deposit into compound
+        uint256 mintStatus = ctoken.mint(amount);
+        require(mintStatus == 0, "compound mint failed");
+
+        // StoGetre ctoken balance after minting
+        uint256 afterBalance = ctoken.balanceOfUnderlying(address(this));
+        // Calculate ctoken shares minted
+        uint256 shares = afterBalance - beforeBalance;
+        // Return the amount of shares the user has produced and the amount of underlying used for it.
+        return (shares, amount);
+    }
+
+    /// @notice Withdraw the number of shares
+    /// @param _shares The number of shares to withdraw
+    /// @param _destination The address to send the output funds
+    // @param _underlyingPerShare The possibly precomputed underlying per share
+    /// @return Amount of funds freed by doing a withdraw
+    function _withdraw(
+        uint256 _shares,
+        address _destination,
+        uint256
+    ) internal override returns (uint256) {
+        // Since ctoken's redeem function returns sucess codes
+        // we get the balance before and after minting to calculate amount
+        uint256 beforeBalance = token.balanceOf(address(this));
+
+        // Do the withdraw
+        uint256 redeemStatus = ctoken.redeem(_shares);
+        require(redeemStatus == 0, "compound redeem failed");
+
+        // Get underlying balance after withdrawing
+        uint256 afterBalance = token.balanceOf(address(this));
+        // Calculate the amount of funds that were freed
+        uint256 amountReceived = afterBalance - beforeBalance;
+        // Transfer the underlying to the destination
+        // 'token' is an immutable in WrappedPosition
+        token.transfer(_destination, amountReceived);
+
+        // Return the amount of underlying
+        return amountReceived;
+    }
+
+    /// @notice Get the underlying amount of tokens per shares given
+    /// @param _amount The amount of shares you want to know the value of
+    /// @return Value of shares in underlying token
+    function _underlying(uint256 _amount)
+        internal
+        override
+        view
+        returns (uint256)
+    {
+        // Load exchange rate
+        uint256 exchangeRate = ctoken.exchangeRateStored();
+
+        // Multiply _amount by exchange rate & correct for decimals
+        // We assume 8 decimals in the ctoken + 18 point decimal fix = 26
+        return ((_amount * exchangeRate) / (10**(26 - underlyingDecimals)));
```

Comment:
I don't think this formula is right, Let's assume cToken is cAave 

Underlying decimals = 18 
cToken decimal = 8
Wrapped position decimals = 18

as per the formula -  amount (i.e wrapped position decimals 18) * exchangeRate ( 1 * 10 ** (18 - 8 + underlying token decimals)) / 10 ** (26 - underlying decimals)
= 18 + 28 - 8 (only calculating the exponents) = still 10 ** 38 remains at the numerator

Let me know if I am missing something here.

Reply:
@smmackey did you take a look into this ?

Reply:
Good catch, this formula has been updated to match the math from the Compound documentation here https://compound.finance/docs#protocol-math